applications/plugins/hid_app/views/hid_mouse_jiggler.c

```diff
@@ -0,0 +1,112 @@
+#include "hid_mouse_jiggler.h"
+#include <gui/elements.h>
+#include "../hid.h"
+
+#include "hid_icons.h"
+
+#define TAG "HidMouseJiggler"
+
+struct HidMouseJiggler {
+    View* view;
+    Hid* hid;
+};
+
+typedef struct {
+    bool running;
+    bool connected;
+} HidMouseJigglerModel;
+
+bool running = false;
+HidMouseJiggler* parentJiggler;
+
+static void hid_mouse_jiggler_draw_callback(Canvas* canvas, void* context) {
+    furi_assert(context);
+    HidMouseJigglerModel* model = context;
+
+    // Header
+    if(model->connected) {
+        canvas_draw_icon(canvas, 0, 0, &I_Ble_connected_15x15);
+    } else {
+        canvas_draw_icon(canvas, 0, 0, &I_Ble_disconnected_15x15);
+    }
+    canvas_set_font(canvas, FontPrimary);
+    elements_multiline_text_aligned(canvas, 17, 3, AlignLeft, AlignTop, "Mouse Jiggler");
+
+    canvas_set_font(canvas, FontPrimary);
+    elements_multiline_text(canvas, AlignLeft, 35, "Press Start\nto jiggle");
+    canvas_set_font(canvas, FontSecondary);
+
+    // Ok
+    canvas_draw_icon(canvas, 63, 25, &I_Space_65x18);
+    if(running) {
+        elements_slightly_rounded_box(canvas, 66, 27, 60, 13);
+        canvas_set_color(canvas, ColorWhite);
+    }
+    canvas_draw_icon(canvas, 74, 29, &I_Ok_btn_9x9);
+    if(running) {
+        elements_multiline_text_aligned(canvas, 91, 36, AlignLeft, AlignBottom, "Stop");
+    } else {
+        elements_multiline_text_aligned(canvas, 91, 36, AlignLeft, AlignBottom, "Start");
+    }
+    canvas_set_color(canvas, ColorBlack);
+
+    // Back
+    canvas_draw_icon(canvas, 74, 49, &I_Pin_back_arrow_10x8);
+    elements_multiline_text_aligned(canvas, 91, 57, AlignLeft, AlignBottom, "Quit");
+
+    if(running) {
+        hid_hal_mouse_move(parentJiggler->hid, MOUSE_MOVE_SHORT, 0);
```

Comment:
View/ViewPort draw call is meant only for drawing, no API calls allowed in its context.
Please rewrite this code in a way when hid_hall only called from application main thread.

Reply:
Thanks for the comment! As I am trying to use the existing HID app, I have not seen any examples where a function can run continously while not locking the main thread and allowing input callbacks.

Is there something like a event loop that can be used in addition to the main thread, or would it be better to move this to its own app and logic?

Reply:
You can use furi_timer and do the work in timer callback

Reply:
Ah fantastic! Thank you!